doc/user-manual/tools/interface-files.rst

```diff
@@ -0,0 +1,47 @@
+.. _interface-files:
+
+********************
+Interface files
+********************
+
+.. note::
+   This is a stub. Contributions, additions and corrections are greatly
+   appreciated.
+
+Probably you already noticed that when you save an agda file, another file
+with the same name and extension .agdai is automatically created. This .agdai
+file is what we call an **interface file** that stores a compressed form of
+the type-checked version of the code that lives in the corresponding .agda file.
+
+Interface files store the results from the type-checking process and resolve
+all implicit arguments (unless you use the flag ``--allow-unsolved-metas``).
+Formally, an .agdai file is a compiled version of the agda source code that can
+be executed by the Agda machine.
+
+Interface files are intended to make operations faster. They translate
+pattern-matching definitions to case trees (this translation speeds up
+computation).
+
+.. note::
+   When an agda file is renamed, the old .agdai file is kept, and a new .agdai
+   file is created. This is the intended behavior, but you can delete the orphan
+   files from your file system if you wish.
+
+Expression trees
+--------------------
+
+An expression tree provides a method of translating source code (in our
```

Comment:
I do not understand this paragraph about expression trees.
Maybe what could be said is:

Type-checked Agda code is internally stored as expression trees.

Reply:
Yes, we could say also this as a shortcut, but still it would be unclear what an expression tree is. I find it useful when possible to explain also the root concepts a bit.

Here, I wanted to give a general idea of what an expression tree is and then how the general concept of expression trees is applied to agda.

I understand that an expression tree is a representation of the source code in a tree-like data structure. Is this correct?

Reply:
Yes.
The formulation "An expression tree provides a method for translating" is confusing, because the tree is not the method but the result of such a translation.

Reply:
Sorry for the delay in my answers, i am multitasking lately more than usual.

OK, i updated the text. Then, what is the name of the method/process that outputs an expression tree from a block of source code?

Reply:
This process is typically called *parsing*, but the documentation for interface files is probably not the best place to talk about this. I think you can mentioned the compression and sharing without talking about expression trees. Maybe just call it the "stored representation of the code" or something like that.